src/main/java/hudson/maven/reporters/SurefireArchiver.java

```diff
@@ -138,7 +138,7 @@ public boolean postExecute(MavenBuildProxy build, MavenProject pom, MojoInfo moj
                 fileSet = Iterables.filter(fileSet, new Predicate<File>() {
                     @Override
                     public boolean apply(File input) {
-                        return !parsedFiles.containsKey(input);
+                        return !parsedFiles.containsKey(input) || parsedFiles.get(input) < input.lastModified();
```

Comment:
I doubt it works reliably in all cases, because the file can be technically replaced by an older one (e.g. move operation). I would expect something like Map<File, Set<Long>>, where the set stores modification times or even MD5s of the files.